idx/memory/memory.go

```diff
@@ -84,6 +82,8 @@ func ConfigSetup() {
 	memoryIdx.DurationVar(&findCacheBackoffTime, "find-cache-backoff-time", time.Minute, "amount of time to disable the findCache when the invalidate queue fills up.")
 	memoryIdx.StringVar(&indexRulesFile, "rules-file", "/etc/metrictank/index-rules.conf", "path to index-rules.conf file")
 	memoryIdx.StringVar(&maxPruneLockTimeStr, "max-prune-lock-time", "100ms", "Maximum duration each second a prune job can lock the index.")
+	memoryIdx.IntVar(&tagquery.MatchCacheSize, "match-cache-size", 1000, "size of regular expression cache in tag query evaluation")
```

Comment:
actually that's what i've done at first, then @dieter asked me to keep the settings in the `memory-idx` (which i think makes sense). It's just unfortunate that now we have to set this setting across packages
https://github.com/grafana/metrictank/pull/1373#discussion_r308388681

Reply:
Is there something else we can do here then? Like keep them completely contained within the index and then after they are loaded initialize them into tagquery with its own variables to hold them? Although then if we ever want to change them at runtime it would be a bit trickier.

It just feels like we are creating a lot of codependency here.

Reply:
Unfortunately the package `tagquery` can't import the package `memory`, this would result in a cycle. We could initialize a setting in `memory`, and then in `ConfigProcess()` basically just copy that value into `tagquery.MetaTagSupport` / `tagquery.MatchCacheSize`. Then both of these two packages would hold a copy of the setting.

Reply:
Yeah something like that.

Reply:
done: https://github.com/grafana/metrictank/pull/1373/commits/aefbc3c0696294448ffc0501b3da0138fc44c354